### PR TITLE
feat: don't allow matcha to skip validation of firm quotes

### PR DESCRIPTION
**Pull request by alexkroeger**
# Description

Validation ensures that the quote provided will be valid for the provided taker.

There is a param that allows a swap quote request to skip validation checks. These changes set this to false always for Matcha.

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -110,6 +110,28 @@ export const getApiKeyWhitelistWithFallback = (
...
 
+export const getApiKeyFromLabel = (
+    label: string,
+): string | undefined => {
+    if (process.env.API_KEYS_ACL === undefined) {
+        return undefined;
+    }
+    let deserialized: ApiKeyStructure;
+    try {
+        deserialized = JSON.parse(process.env.API_KEYS_ACL);
```
Comment (phil-ociraptor) [inline src/config.ts:121]: Is there a way we could avoid paying the deserialization cost on every call to this function? 

Maybe just make this object value an exported constant or something

Comment (phil-ociraptor) [reply src/config.ts:121]: We could actually generate two lookup maps at start: 

- One by api key (which is the default)
- Another map by the label value

Comment (alexkroeger) [reply src/config.ts:121]: that's a good call, I think we'd only call this function once on startup tho

**Review by phil-ociraptor — APPROVED** [review]

**Comment by alexkroeger** [comment]
deploy staging

**Review by PirosB3 — APPROVED** [review]

**Pull request merged**